python/ray/autoscaler/_private/resource_demand_scheduler.py

```diff
@@ -38,19 +39,21 @@
 
 
 class ResourceDemandScheduler:
-    def __init__(self, provider: NodeProvider,
-                 node_types: Dict[NodeType, NodeTypeConfigDict],
-                 max_workers: int):
+    def __init__(self, provider: NodeProvider, cluster_config: Dict[str, Any]):
```

Comment:
Please revert this to the config values explicitly instead of an entire config in one big dict. This reduces the coupling between components.

Reply:
but when we will remove min_workers for multi node types this can be an issue. Unless we do:
`config.avalable_node_types[TAG_RAY_NODE_TYPE_LEGACY] = config["worker_nodes"]`
in `prepare_config`

Reply:
Why is it an issue, isn't this just a matter of passing the dependencies individually as separate arguments instead of a big dict?